### DISCUSSION TITLE
Ark module (non-custodial Fedimints)

### DISCUSSION THREAD

**Discussion by jkitman** [fedimint/fedimint#2631] [Fedimint Modules]
Recently [Ark](https://www.arkpill.me/) has been proposed as a layer 2 solution requiring a centralized but non-custodial coordinator.

One of the major advantages of Ark is it removes the need for new users to have inbound liquidity on Lightning.  Interestingly, Fedimint solves this too by trading-off the on-chain fees of Ark for federated custodial-risk.

The major disadvantages seem to be:
- Clients need to come online at least once every 4 weeks or will fall back to the federated trust model
- Federation needs to have enough on-chain balance to support all the Ark txs expected within a 4-week period
- Federation needs to eat the costs of one tx every 5 sec (the "settlement" interval)
- Receivers of payments must be online until covenants
- Smaller anonymity sets (whoever is online during the 5-second tx interval)

Ark could allow Fedimint to have a non-custodial module, while at the same time federating the coordinator could help prevent DoS style attacks.  @moonsettler might have some more insights on this.

Since Ark is still in the early spec phases I wouldn't expect us to be able to implement anything in Fedimint for a while.

**Comment by davidohayon669** [comment]
it seems like ark solves the atomic swapping of vTXOs elegantly (like some other covenants models) but i'm afraid it's trying to be too perfect by handing all transaction granularities. the 5 seconds interval introduces new bottlenecks and the dust vTXOs makes no sense to eject to base layer. covenants vTXOs are great for savings but less for payments. this is where the federation model can find a perfect fit

imagine a group of 5-10 anon LN node operators pool their resources together to offer a hybrid model between ecash and vTXOs. the savings portion of a user (80-95%) will be kept in these vTXOs that pay a fee for the rolling cost to the fed (the base layer txn and the cost of capital if you used the vTXO before it "matured"). this also removes the rigidity of the 4 weeks duration - if a user knows he's not gonna touch some funds for a year, why not put them in a cheaper 365 days vTXO, which also requires less interactivity? the "checking account" (5-20%) of a user's funds can be handled with ecash. a good balance between earning fees on funds that cannot be rugged AND the routing fees for ecash payments can lead to very good market incentives for not cheating even between anonymous parties. as long as MOST funds are non-ruggable, we can utilize the simplicity and speed of ecash and the security of vTXOs handled by the same federation of providers

more then that, vTXO funds don't necessarily need to be handled by the fed - these can be serviced by each guardian independently, like LN gateways. the only assets a federation should share are the ecash wallet funds. guardians that provide vTXOs can hold some buffer funds in ecash (same as for LN gateways) so that the movement between ecash and vTXOs don't require too much on-chain txns

**Reply by humansinstitute** [reply]
I quite like this idea.

**Comment by humansinstitute** [comment]
I really quite like Ark, I do have a concern about how "usefully non-custodial" this model would be over a longer time frame with broad adoption of bitcoin.  

The main concern being that I don't see how I could ever get a truly non-custodial experience on bitcoin unless I have the  economic capacity to actually own an on chain UTXO - which I fully expect to be priced out for most people on the planet. 

Logic would be: 

- Broad adoption of bitcoin fills mempool, UTXOs capacity.  
- We quickly price out transactions on chain that don't represent some massive batching off-chain
- Fees run to $1000's (in current purchasing power equivalent)
- We now can't meaningfully "self custody" anything less than this limit as the fee eats it. 

An Ark module would be cool to hold and transact with vTXOs of lower value, but I'm not sure you'd get the non-custodial benefit if you can't meaningfully "fall back on chain" - unless I'm missing something here? 

Perhaps there's a more - mutually assured destruction approach to custody here where you can't get out on chain so it's no non-custodial, but there is a middle ground where you can't get out on chain (because of fee issues) but you can also stop guardians from rugging you.

**Reply by moonsettler** [reply]
mostly agreed. what the ASP can do in regards to mutually assured destruction is to pad the small amount vTXOs with most of the expected on-chain fees. someone pushing to chain would still lose sats so the loss minimizing strategy is to cooperate. Ark can also straddle chains, assuming CTV is activated surfchains maintained by ASPs (specific type of spacechain) can be used as a subsatoshi feerate alternative to mainchain for near and below dust vTXOs.

**Reply by humansinstitute** [reply]
I hadn't heard of surfchains - new S-Chain dropped whilst I was looking the other way need to catch up! 

So would I be in a Surfchain ASP at all times and the protocoll just rundson both chains, or the ASP could be "on bitcoin", but redeem in a conflict to the surfchain?

**Reply by moonsettler** [reply]
you want to be on the most secure chain that you can afford or makes sense for the amounts. lemme drop the link...
so the idea is you would have large value vTXOs on mainchain do cross chain swaps when you need to and have the "change" on a sidechain with low fees where those sats are economically enforceable. it also saves a lot of liquidity for the ASP.
https://gist.github.com/moonsettler/e0c24863fda418a2fb504a307fda8a19

**Reply by moonsettler** [reply]
btw the main drawback for surfchains is they are by design impermanent. they exist only till there is transactional demand. but this is not a problem as far as Ark is concerned as Ark pool transactions have an even shorter lifespan and Ark pools have a good predictable baseline demand for block space. so these might mesh exceptionally well.

**Reply by humansinstitute** [reply]
> you want to be on the most secure chain that you can afford or makes sense for the amounts. lemme drop the link... so the idea is you would have large value vTXOs on mainchain do cross chain swaps when you need to and have the "change" on a sidechain with low fees where those sats are economically enforceable. it also saves a lot of liquidity for the ASP. https://gist.github.com/moonsettler/e0c24863fda418a2fb504a307fda8a19

That makes a lot of sense. Kind of reminds me of Eric Voskuils old ideas of horizontally scaling bitcoin across different chains that have different security guarantees for lower denominations.

**Comment by humansinstitute** [comment]
> the main drawback for surfchains is they are by design impermanent

Is there a good place to go to catch up on this, or is it just searching through the Spacechains telegram?

**Reply by moonsettler** [reply]
i don't actually know the specifics of what Ruben Somsen and his crew are working on. the concept of surfchain came from a discussion whether drivechains can or can not be implemented via CTV. i said no, and tried to show what the closest you can come to a two way peg.
it's obviously lacking compared to a real two way peg. the name came from the idea that liquidity is added in huge waves and gets rode out. a bit like surfing.

(btw credit for the general idea goes to Ruben Somsen he mentioned burning spacechain tokens to maintain a peg in a discussion)

**Comment by BritvaBo** [comment]
Interesting dialogue guys.
